internal/activity.go

```diff
@@ -264,6 +264,13 @@ func RecordActivityHeartbeat(ctx context.Context, details ...interface{}) {
 	getActivityOutboundInterceptor(ctx).RecordHeartbeat(ctx, details...)
 }
 
+// GetClient returns a client that can be used to interact with the Temporal
+// service from an activity.
+func GetClient(ctx context.Context) Client {
+	activityEnv := getActivityEnv(ctx)
+	return activityEnv.serviceInvoker.GetClient(ClientOptions{Namespace: activityEnv.workflowNamespace})
```

Comment:
hm, but you are not using the client , or client options the worker was started with. So if the worker client was configured with a data converter or interceptor I don't think this client will have it.

Reply:
I think you might need to just wire the client from worker constructor down to here.